src/App.js

```diff
@@ -1,8 +1,27 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
 
 const App = () => {
+
+  const [chatEntries, setChatsLike] = useState(chatMessages)
+
+  const updateLikeData = updatedChat =>{
+    const chats = chatEntries.map(chat => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      }else {
+        return chat;
+      }
+    }); 
+
+    setChatsLike(chats)
```

Comment:
👀 Since the next version of the chat data depends on the current version, prefer using the callback style of `setChatsLike`
```js
    setChatsLike(chats => (
      // logic to use current chats to make a new copy with the desired liked status flipped (the map logic above)
      // return the new chats to update the state
    ))
```